src/integrators/integrator_utils.jl

```diff
@@ -107,3 +107,252 @@ end
     bool, saved_in_cb2 = apply_discrete_callback!(integrator, ts, us, callback)
     discrete_modified || bool, saved_in_cb || saved_in_cb2
 end
+
+@inline function interpolate(integrator, t)
+    θ = (t - integrator.tprev) / integrator.dt
+    b1θ, b2θ, b3θ, b4θ, b5θ, b6θ, b7θ = SimpleDiffEq.bθs(integrator.rs, θ)
+    return integrator.uprev +
+           integrator.dt *
+           (b1θ * integrator.k1 + b2θ * integrator.k2 + b3θ * integrator.k3 +
+            b4θ * integrator.k4 + b5θ * integrator.k5 + b6θ * integrator.k6 +
+            b7θ * integrator.k7)
+end
+
+@inline function _change_t_via_interpolation!(integrator, t,
+                                              modify_save_endpoint::Type{Val{T}}) where {T}
+    # Can get rid of an allocation here with a function
+    # get_tmp_arr(integrator.cache) which gives a pointer to some
+    # cache array which can be modified.
+    if integrator.tdir * t < integrator.tdir * integrator.tprev
+        error("Current interpolant only works between tprev and t")
+    elseif t != integrator.t
+        integrator.u = integrator(t)
+        integrator.t = t
+        #integrator.dt = integrator.t - integrator.tprev
+    end
+end
+@inline function DiffEqBase.change_t_via_interpolation!(integrator::Union{
+                                                                          GPUTsit5Integrator,
+                                                                          GPUATsit5Integrator
+                                                                          }, t,
+                                                        modify_save_endpoint::Type{Val{T}} = Val{
+                                                                                                 false
+                                                                                                 }) where {
+                                                                                                           T
+                                                                                                           }
+    _change_t_via_interpolation!(integrator, t, modify_save_endpoint)
+end
+
+@inline function apply_callback!(integrator,
+                                 callback::GPUContinuousCallback,
+                                 cb_time, prev_sign, event_idx)
+    DiffEqBase.change_t_via_interpolation!(integrator, integrator.tprev + cb_time)
+
+    saved_in_cb = true
+
+    integrator.u_modified = true
+
+    if prev_sign < 0
+        if callback.affect! === nothing
+            integrator.u_modified = false
+        else
+            callback.affect!(integrator)
+        end
+    elseif prev_sign > 0
+        if callback.affect_neg! === nothing
+            integrator.u_modified = false
+        else
+            callback.affect_neg!(integrator)
+        end
+    end
+
+    true, saved_in_cb
+end
+
+@inline function handle_callbacks!(integrator::GPUATsit5Integrator)
```

Comment:
Why does this one exist?

Reply:
This exists because we need `ts, us` in the Unadaptive case to save the solution before applying it to the callback in the discrete case. Otherwise, it regresses from the `OrdinaryDiffEq` implementation. Hence, I thought to make to another dispatch for the adaptive version in which we aren't saving values in between, else I could have handled with if-else in one `handle_callbacks!` (which would happen at runtime)

Reply:
Since `adaptive = false` can only work with `save_everystep = false`, It does not need to save to `ts,us`. https://github.com/SciML/JumpProcesses.jl/blob/master/src/SSA_stepper.jl#L278

Reply:
Anyways, I made it to dispatch to single `handle_callbacks!`